.github/workflows/build-container.yml

```diff
@@ -36,6 +36,10 @@ jobs:
       with:
         path: /tmp/.buildx-cache
         key: base-${{ matrix.repo }}-buildx-${{ github.sha }}
+        # allow cache hits from previous runs of the current branch,
+        # parent branch, then upstream branches, in that order
+        restore-keys: |
+          base-${{ matrix.repo }}-buildx-
```

Comment:
Not familiar with restore-keys... do you need a `${{ github.sha }}` at the end? What creates this key?

Reply:
These docs do a pretty good job of explaining the cache key hit behavior: https://docs.github.com/en/actions/guides/caching-dependencies-to-speed-up-workflows#matching-a-cache-key

tl;dr This `actions/cache` step creates the SHA-specific key and the restore-key `base-${{ matrix.repo }}-buildx-` allows for cache hits that aren't SHA-specific. So excluding the Git SHA allows this step to load the buildx cache from previous GHA runs, speeding up subsequent builds.

Reply:
OK, makes sense.